src/lua/widget/container.c

```diff
@@ -154,16 +154,35 @@ static int container_numindex(lua_State*L)
       GtkWidget *searched_widget = g_list_nth_data(children,index);
       gtk_container_remove(GTK_CONTAINER(container->widget),searched_widget);
     } else {
-      luaL_error(L,"Incorrect index or value when setting the child of a container : you can only append, not change a child in the middle");
+      GList *current;
+      for(current = g_list_nth(children, index); current != NULL; current = current->next)
+      {
+        gtk_container_remove(GTK_CONTAINER(container->widget), current->data);
```

Comment:
`gtk_container_remove()` may destroy the widget, so call `g_object_ref()` on the widget first.